modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4MessageChannelHandler.java

```diff
@@ -131,11 +128,6 @@ private void doFlush(ChannelHandlerContext ctx) {
                 break;
             }
             final WriteOperation write = currentWrite;
-            if (write.buf.readableBytes() == 0) {
```

Comment:
Should we assert that `write.buf.readableBytes() > 0`?

Reply:
Hmm technically we could and it would hold today but I'm -0 on it I think. Technically writing a zero length message should be possible at this step of the pipeline and there isn't really a reason to treat zero length writes in any special way nor do we rely on the fact that the buffers aren't zero length here.